### PR TITLE
Refined timeout conditions (#785)

### PR DESCRIPTION
**Pull request by GPelayo**
IntegrationTest.wait_for_queue_level() now stops waiting if the last 10 recorded queue states have an average size of 0.

IntegrationTest.check_bundles_are_indexed() fails if 10 minutes has passed since the first api call to the service and all the test bundles aren't found. It no longer fails after checking all the pages of the service 5 times.

Improved logs and import order.

### CODE REVIEW

File: test/local_integration_test.py [hunk 1]
```diff
@@ -129,24 +129,30 @@ def wait_for_queue_level(self, empty: bool = True, timeout: int = 60):
         queues = {queue_name: boto3.resource('sqs').get_queue_by_name(QueueName=queue_name)
                   for queue_name in queue_names}
         wait_start_time = time.time()
+        queue_size_history = deque(maxlen=10)
 
         while True:
             total_message_count = self.get_number_of_messages(queues)
             queue_wait_time_elapsed = (time.time() - wait_start_time)
+            queue_size_history.append(total_message_count)
+            average_queue_size = sum(queue_size_history) / queue_size_history.maxlen
```
Comment (hannes-ucsc) [inline test/local_integration_test.py]:
```suggestion
            cumulative_queue_size = sum(queue_size_history)
```

File: test/local_integration_test.py [hunk 2]
```diff
@@ -129,24 +129,30 @@ def wait_for_queue_level(self, empty: bool = True, timeout: int = 60):
...
             if queue_wait_time_elapsed > timeout:
                 logger.error('The queue(s) are NOT at the desired level.')
                 return
-            elif (total_message_count <= 0) == empty:
+            elif (average_queue_size == 0) == empty:
```
Comment (hannes-ucsc) [inline test/local_integration_test.py]:
```suggestion
            elif (cumulative_queue_size == 0) == empty:
```

File: test/local_integration_test.py [hunk 3]
```diff
@@ -129,24 +129,30 @@ def wait_for_queue_level(self, empty: bool = True, timeout: int = 60):
...
                 logger.info('The queue(s) at the desired level.')
                 break
+            else:
+                logger.info('The average size of the last %i recorded queue states is %.2f.',
```
Comment (hannes-ucsc) [inline test/local_integration_test.py]:
```suggestion
                logger.info('The most recently sampled queue sizes are %r',
```

File: test/local_integration_test.py [hunk 4]
```diff
@@ -129,24 +129,30 @@ def wait_for_queue_level(self, empty: bool = True, timeout: int = 60):
...
+                            queue_size_history.maxlen, average_queue_size)
```
Comment (hannes-ucsc) [inline test/local_integration_test.py]:
```suggestion
                            queue_size_history)
```

File: test/local_integration_test.py [hunk 5]
```diff
@@ -100,24 +100,29 @@ def wait_for_queue_level(self, empty: bool = True, timeout: int = 60):
...
+            cumulative_queue_size = sum(queue_size_history)
             if queue_wait_time_elapsed > timeout:
                 logger.error('The queue(s) are NOT at the desired level.')
                 return
-            elif (total_message_count <= 0) == empty:
+            elif (cumulative_queue_size == 0) == empty:
                 logger.info('The queue(s) at the desired level.')
                 break
+            else:
+                logger.info('The most recently sampled queue sizes are %r.', cumulative_queue_size)
```
Comment (hannes-ucsc) [inline test/local_integration_test.py]:
```suggestion
                logger.info('The most recently sampled queue sizes are %r.', queue_size_history)
```

I did [ask](https://github.com/DataBiosphere/azul/pull/793#discussion_r265741718) for this in the last review.

Comment (GPelayo) [reply test/local_integration_test.py]: Sorry, I misread the change.

**Pull request merged**